setup.py

```diff
@@ -54,6 +54,7 @@
         "colorlog>=2.6.1,<5.0.0",
         "py>=1.4.0,<2.0.0",
         "virtualenv>=14.0.0",
+        "requests",
```

Comment:
I am very against adding new dependencies in nox. Can we think of a way to do this without taking on new dependencies?

Reply:
I was afraid of this :)
well, there are two paths here:

 - either we remove `requests` from the dependencies BUT we try to import it when the offline-checking procedure starts, and we do not find it we fallback to the legacy behaviour (no online/offline checks)

 - or we write the equivalent code using stdlib only. The "only" issue with this is that it might be hard to get a complete equivalent feature supporting http proxy settings and other complexities that `requests` handle for us.

Since `requests` is available in most python environments today I would tend to prefer option 1 as the risk of doing the second one wrong is high. What do you think?

Reply:
Actually I'll still give it a go as apart from proxies it seems quite ok. Let me try